File: game_plugins/hase_und_igel_new/server/sc/plugin2018/GamePlugin.java
package sc.plugin2018;

import sc.api.plugins.IGameInstance;
import sc.api.plugins.IGamePlugin;
import sc.api.plugins.host.IGamePluginHost;
import sc.plugin2018.util.Configuration;
import sc.plugins.PluginDescriptor;
import sc.shared.ScoreAggregation;
import sc.shared.ScoreDefinition;
import sc.shared.ScoreFragment;

/**
 * Die Beschreibung des Hase- und Igel Core-Plugins für die Software-Challenge
 * 2018.
 *
 */
@PluginDescriptor(name = "Hase und Igel", uuid = GamePlugin.PLUGIN_UUID, author = GamePlugin.PLUGIN_AUTHOR)
public class GamePlugin implements IGamePlugin
{

	public static final String			PLUGIN_AUTHOR		= "";
	public static final String			PLUGIN_UUID			= "swc_2018_hase_und_igel";

	public static final ScoreDefinition	SCORE_DEFINITION;

	static
	{
		SCORE_DEFINITION = new ScoreDefinition();
		SCORE_DEFINITION.add("Gewinner");
		// NOTE: Always write the XML representation of unicode characters, not the character directly, as it confuses the
		// parsers which consume the server messages!
		SCORE_DEFINITION.add(new ScoreFragment("\u2205 Feldnummer",
				ScoreAggregation.AVERAGE));
		SCORE_DEFINITION.add(new ScoreFragment("\u2205 Karotten",
				ScoreAggregation.AVERAGE));
	}

	@Override
	public IGameInstance createGame()
	{
	  return new Game(PLUGIN_UUID);
	}

	@Override
	public void initialize(IGamePluginHost host) {
		host.registerProtocolClasses(Configuration.getClassesToRegister());
	}

	@Override
	public void unload() {
		// TODO Plugin entladen
	}

	@Override
	public ScoreDefinition getScoreDefinition()
	{
		return SCORE_DEFINITION;
	}

}

File: plugin/src/server/sc/plugin2021/GamePlugin.kt
package sc.plugin2021

import sc.api.plugins.IGameInstance
import sc.api.plugins.IGamePlugin
import sc.api.plugins.IGameState
import sc.api.plugins.host.GameLoader
import sc.plugins.PluginDescriptor
import sc.shared.ScoreAggregation
import sc.shared.ScoreDefinition
import sc.shared.ScoreFragment
import java.io.File

@PluginDescriptor(name = "Blokus", uuid = GamePlugin.PLUGIN_UUID)
class GamePlugin: IGamePlugin {
    companion object {
        const val PLUGIN_UUID = "swc_2021_blokus"
    }
    
    override fun id() = PLUGIN_UUID
    
    override fun createGame(): IGameInstance =
            Game()
    
    override fun createGameFromState(state: IGameState): IGameInstance =
            Game(state as GameState)
    
    override val scoreDefinition: ScoreDefinition =
            ScoreDefinition(arrayOf(
                    ScoreFragment("Gewinner"),
                    ScoreFragment("\u2205 Punkte", ScoreAggregation.AVERAGE)
            ))
    
}
